packages/@aws-cdk/pipelines/lib/codepipeline/codebuild-step.ts

```diff
@@ -144,10 +165,28 @@ export class CodeBuildStep extends ShellStep {
    */
   readonly securityGroups?: ec2.ISecurityGroup[];
 
+  /**
+   * Use on source code which is not going to be the synth step, must also provide a `fullBuildSpec`.
+   * This will not use `commands`, `installCommands` or `partialBuildSpec` it will only reference `fullBuildSpec`, if provided.
+   * @default - false
+   */
+  readonly skipDefaultBuildSpec?: boolean
+
+  /**
+    * Use to provide a full build spec which can reference a buildspec on disk.  Must also specify `skipDefaultBuildSpec` as `true`.
+    * @default - undefined will not use a build spec in the build project.
+    */
+  readonly fullBuildSpec?: codebuild.BuildSpec;
```

Comment:
I kinda don't like having both the  `fullBuildSpec` AND `skipDefaultBuildSpec` here. In my mind, we have the following options:

* Just add `fullBuildSpec` and have it be mutex with `partialBuildSpec`
* Have `partialBuildSpec` and `autoBuildSpec` -- can't find a good name for this. Needs to encompass both the automatic addition of artifacts as well as the inheriting of defaults from the pipeline.
* Have `buildSpec and `buildSpecIsComplete` -- as before but deprecating `partialBuildSpec` for a new `buildSpec` parameter.

None of these are obviously great, but before we make any decisions here I want to talk use cases first.